### PR TITLE
Upgrade to pac4j 6.x, require Java 17 or newer, and migrate to EE 9

### PR DESCRIPTION
**Pull request by basil**
https://www.pac4j.org/blog/what_s_new_in_pac4j_v6.html

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env groovy
 
-buildPlugin(useContainerAgent: true, configurations: [
+buildPlugin(useContainerAgent: false, configurations: [
```
Comment (basil) [inline Jenkinsfile:3]: To run `LiveTest` on CI builds.

File: src/main/java/org/jenkinsci/plugins/saml/SamlProfileWrapper.java [hunk 2]
```diff
@@ -48,13 +50,16 @@ public SamlProfileWrapper(SamlPluginConfig samlPluginConfig, StaplerRequest requ
...
         SAML2Profile saml2Profile;
         try {
             SAML2Client client = createSAML2Client();
             WebContext context = createWebContext();
-            credentials = (SAML2Credentials) client.getCredentials(context, JEESessionStore.INSTANCE).orElse(null);
-            saml2Profile = (SAML2Profile) client.getUserProfile(credentials, context, JEESessionStore.INSTANCE).orElse(null);
+            SessionStore sessionStore = createSessionStore();
+            CallContext ctx = new CallContext(context, sessionStore);
+            SAML2Credentials unvalidated = (SAML2Credentials) client.getCredentials(ctx).orElse(null);
+            credentials = (SAML2AuthenticationCredentials) client.validateCredentials(ctx, unvalidated).orElse(null);
```
Comment (basil) [inline src/main/java/org/jenkinsci/plugins/saml/SamlProfileWrapper.java:61]: Per [the migration guide](https://www.pac4j.org/blog/what_s_new_in_pac4j_v6.html), the `getCredentials` method has been split into the `getCredentials` and `validateCredentials` methods.

File: src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java [hunk 3]
```diff
@@ -596,30 +597,30 @@ static String getSPMetadataFilePath() {
...
     /**
-     * @see SecurityRealm#getPostLogOutUrl
+     * @see SecurityRealm#getPostLogOutUrl2
      * Note: It does not call the logout service on SAML because the library does not implement it on this version,
      * it will be done when we upgrade the library.
      */
     @SuppressWarnings("deprecation")
     @Override
-    protected String getPostLogOutUrl(StaplerRequest req, @NonNull Authentication auth) {
+    protected String getPostLogOutUrl2(StaplerRequest2 req, @NonNull Authentication auth) {
```
Comment (basil) [inline src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java:611]: Migrating from Acegi to Spring Security.

**Review by kuisathaverat — APPROVED** [review]

**Comment by kuisathaverat** [comment]
It will trigger a release. Thanks for the changes.

File: src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java [hunk 4]
```diff
@@ -596,30 +597,30 @@ static String getSPMetadataFilePath() {
...
-    public HttpResponse doMetadata(StaplerRequest request, StaplerResponse response) {
+    public HttpResponse doMetadata(StaplerRequest2 request, StaplerResponse2 response) {
         return new SamlSPMetadataWrapper(getSamlPluginConfig(), request, response).get();
     }
 
     /**
-     * @see SecurityRealm#getPostLogOutUrl
+     * @see SecurityRealm#getPostLogOutUrl2
      * Note: It does not call the logout service on SAML because the library does not implement it on this version,
      * it will be done when we upgrade the library.
```
Comment (basil) [inline src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java:607]: @kuisathaverat Now that the library has been upgraded, this task can be completed. From [the blog post](https://www.pac4j.org/blog/what_s_new_in_pac4j_v6.html):

> The `processLogout` method has been added, based on the `LogoutProcessor` component.

Comment (kuisathaverat) [reply src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java:607]: I have limited time, I do not plan to do it soon

**Pull request merged**